includes/traits/llms-trait-award-templates-post-list-table.php

```diff
@@ -80,23 +80,37 @@ public function add_post_actions( $actions, $post ) {
 			)
 		);
 
-		$engagement_type = str_replace( 'llms_', '', $post->post_type );
-		$sync_action     = "sync_awarded_{$engagement_type}s";
-		$sync_url        = add_query_arg(
-			array(
-				'action'                                      => $sync_action,
-				"_llms_{$engagement_type}_sync_actions_nonce" => wp_create_nonce( "llms-{$engagement_type}-sync-actions" ),
-			),
-			get_edit_post_link( $post, 'raw' )
-		);
+		$awarded_number = $this->count_awarded_engagements( $post->ID );
```

Comment:
don't you think that running this query for each and every template in the list might be a bit too much heavy?
I did consider doing the same when adding the "View Awarded *" link, see: https://github.com/gocodebox/lifterlms/pull/1985#issue-1128842305

I think would be better to run the check when you click on the link. I mean your solution is cleaner, but I think it might be pretty heavy for no real reason. What do you think?

Reply:
You have a good point @eri-trabiccolo.

Every template should have a "Sync Awarded Xs" post action.
When there are no awarded engagements to sync, display an admin notice. Something like 'X Template "NAME" (ID) has no awarded Xs to sync with.'

Is this what you had in mind?

Reply:
Yeah. That's what I had in mind. <3